core/templates/components/entity-creation-services/topic-creation.service.ts

```diff
@@ -67,10 +124,16 @@ angular.module('oppia').factory('TopicCreationService', [
           ]
         });
 
-        modalInstance.result.then(function(topic) {
-          if (topic.topicName === '') {
+        modalInstance.result.then(function(topicObject) {
```

Comment:
Use a domain object throughout, and call it the name of the domain object. For example, if it is a Topic domain object, then have the variable name be "topic". If it's something else, change the variable name accordingly. We typically do not add the "Object" suffix to variable names.

Reply:
Done.